AWSClientRuntime/Sources/Regions/DefaultRegionResolver.swift

```diff
@@ -5,26 +5,32 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 import AwsCommonRuntimeKit
+import ClientRuntime
 
 public struct DefaultRegionResolver: RegionResolver {
     public let providers: [RegionProvider]
-    
+    let logger: SwiftLogger
+
     public init(providers: [RegionProvider] = [EnvironmentRegionProvider(), ProfileRegionProvider(), IMDSRegionProvider()]) {
         // TODO: add more region resolvers i.e. System Properties, etc
         self.providers = providers
+        self.logger = SwiftLogger(label: "DefaultRegionProvider")
     }
     
     public func resolveRegion() -> String? {
         for provider in providers {
             do {
                 // TODO: Verify that this does not block calling thread. If it does, we need to solve this similar to the way we solve this for the credentials provider
+                logger.debug("Attempting to resolve region with: \(String(describing: type(of: provider)))")
                 if let region = try provider.resolveRegion().get() {
+                    logger.debug("Resolved region with: \(String(describing: type(of: provider)))")
                     return region
                 }
             } catch {
                 return nil
             }
         }
+        logger.error("Unable to resolve region")
```

Comment:
@kneekey23  -- do you know of any cases where returning nil isn't an error?

Reply:
returning nil is never an error. user could provide a region either at the client level or later on at the operation level. this default chain is basically to get a region if they didnt provide it. but they should be able to even make a request without providing a region and then see an error

Reply:
love all these changes minus this one. can we just do logger.debug here or logger.info instead of error? it's not an error if this region resolver didn't find a region, this is a nicety the sdk provides so they dont have to give us a region. and in certain cases like lambda or ec2, they shouldnt have to at all.

Reply:
Updated to `.debug`

Reply:
FYI -- this means that case1 and case4 (see above) output nothing from the SDK -- (i think this is the behavior we want)

Reply:
ya I agree, its confusing as .info at least some customers keep getting confused thinking it's an error when it's not. looks good to me.